### ISSUE TITLE
[API] About batch_normalization

### ISSUE THREAD

**Issue by zhoonit** [nnstreamer/nntrainer#209]
AFAIK, batch normalization is a process should be done **before** activation happens for every layer except the last layer.

So I think, in the public api level, `bn layer` can be hidden. Rather, it can be property of either `layer` or `Network`

eg) if you want to make a bn layer.

current version requires to declare a layer.

```ini
# Network Section : Network
[Network]
Type = NeuralNetwork	# Network Type : Regression, KNN, NeuralNetwork
Layers = inputlayer \
         fc1layer \
         batchnormalization \
         fc2layer \
         batchnormalization2 \
	 outputlayer	#Layers of Neuralnetwork
# /** omitted **/

# Layer Section : Name
[inputlayer]
Type = InputLayer

[fc1layer]
Type = FullyConnectedLayer

[fc2layer]
Type = FullyConnectedLayer

[batchnormalization]
Type = BatchNormalizationLayer

[batchnormalization2]
Type = BatchNormalizationLayer

[outputlayer]
Type = OutputLayer

```

but this could be more concise if we put `bn_layer` as a properties like: 

```ini
# Network Section : Network
[Network]
Type = NeuralNetwork	# Network Type : Regression, KNN, NeuralNetwork
Layers = inputlayer \
         fc1layer \
         fc2layer \
	 outputlayer	#Layers of Neuralnetwork
# /** omitted **/

# Layer Section : Name
[inputlayer]
Type = InputLayer

[fc1layer]
Type = FullyConnectedLayer
BatchNormalize = true

[fc2layer]
Type = FullyConnectedLayer
BatchNormalize = true

[outputlayer]
Type = OutputLayer
```

It would be more concise if we assume that bn follows just before every activation happens. 
Since we can make it as a `Network` properties:  

```ini
# Network Section : Network
[Network]
Type = NeuralNetwork	# Network Type : Regression, KNN, NeuralNetwork
Layers = inputlayer \
         fc1layer \
         fc2layer \
	 outputlayer	#Layers of Neuralnetwork
BatchNormalize = true
# /** omitted **/

# Layer Section : Name
[inputlayer]
Type = InputLayer

[fc1layer]
Type = FullyConnectedLayer

[fc2layer]
Type = FullyConnectedLayer

[outputlayer]
Type = OutputLayer
```

My thinking is that **network should have the `bn` property** and we handle it for the user, providing that cases which violating the rule(batch normal and then activation for every layer except the last one) are rare.

I'd like to hear opinions. @jijoongmoon  @kparichay

**Comment by kparichay** [comment]
@zhoonit Overall I agree to having batch normalization as a property for an individual layer (just like activation). Its more cleaner and inline with existing frameworks style.
However, the constraint `bn follows just before every activation happens` is not necessary. How about adding another ini config `batch_normalize_after_activation` which is false by default?

**Comment by zhoonit** [comment]
> @zhoonit Overall I agree to having batch normalization as a property for an individual layer (just like activation). Its more cleaner and inline with existing frameworks style.
> However, the constraint `bn follows just before every activation happens` is not necessary. How about adding another ini config `batch_normalize_after_activation` which is false by default?

Sounds good idea. I will work to make bn as a layer property first, and we could extend with `batch_normalize_after_activation` prop
